### DISCUSSION TITLE
The browser cannot receive cookies from FastAPI set_cookie methods

### DISCUSSION THREAD

**Discussion by seyong92** [fastapi/fastapi#6057] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post('/login', summary='Create access and refresh tokens for user', response_model=TokenSchema)
async def login(form_data: OAuth2PasswordRequestForm = Depends()):
    ...

    response = JSONResponse({'access_token': create_access_token(user['id'])})
    response.set_cookie(key='refresh_token_test', value=create_refresh_token(user['id']),
                        max_age=REFRESH_TOKEN_EXPIRE_MINUTES, httponly=False, samesite='none', domain='http://localhost:3000')

    return response
```

### Description

Currently, I am using a backend with FastAPI (port 8000) and a frontend with Solid-JS (port 3000).

I want to send a refresh token from the backend server to the client when they log in.

For login, I send a request from the client using Axios like below:

```typescript
const onClickLogin = () => {
    axios({
      method: 'post',
      url: 'http://localhost:8000/login',
      responseType: 'json',
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      },
      data: {
        username: inputUsername(),
        password: inputPw(),
      },
    }).then((response) => {
      props.setToken(response.data.access_token);
      props.updateUserinfo();
      props.setPageStatus('loggedin');
    });
  };
```

In this case, I just disabled 'httponly' option to check cookie more easily in Chrome developer tools.

It is very difficult to check what is the problem because the response is received successfully and does not return any error or warnings but there are just no cookies.

I also set the CORS setting in FastAPI like below.

```python
origins = [
    'http://localhost:3000'
]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)
```

Is there any reasons that the browser does not receive cookies? Is there any method to debug this case?

### Operating System

macOS

### FastAPI Version

0.88.0

### Python Version

3.10.5

**Comment by iudeen** [comment]
Can you attach the XHR?

**Comment by zoola969** [comment]
@seyong92 According to security policy you have to set `secure=True` for cookies with `samesite="none"`.  
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite
